src/govuk/components/pagination/template.njk

```diff
@@ -0,0 +1,76 @@
+{% set type = params.type | default("results") %}
+{% set blockLevel = not params.items and params.next and params.previous %}
+
+
+<nav class="govuk-pagination {{- ' ' + params.classes if params.classes}} {{'govuk-pagination--block' if blockLevel }}" role="navigation" aria-label="{{type}}">
+  <ul class="govuk-pagination__list">
+    {% if params.previous and params.previous.href %}
+      <li class="govuk-pagination__item govuk-pagination__item--prev">
+        <a class="govuk-link govuk-pagination__link" href="{{ params.previous.href }}" rel="prev">
+          <span class="govuk-pagination__link-title {{'govuk-pagination__link-title--decorated' if blockLevel and not params.previous.label}}">
+            <svg class="govuk-pagination__icon govuk-pagination__icon--prev" xmlns="http://www.w3.org/2000/svg" height="13" width="15" aria-hidden="true" focusable="false" viewBox="0 0 15 13">
+              <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
+            </svg>
+            {% if params.previous.text %}
+              {{ params.previous.text }}
+            {% else %}
+              Previous <span class="govuk-visually-hidden">page</span>
+            {% endif %}
+          </span>
+
+          {% if params.previous.label %}
+            <span class="govuk-visually-hidden">:</span>
+            <span class="govuk-pagination__link-label">{{ params.previous.label }}</span>
+          {% endif %}
+        </a>
+      </li>
+    {% endif %}
+
+    {% for item in params.items %}
+      {% if item.ellipsis %}
+        <li class="govuk-pagination__item govuk-pagination__item--ellipses">&ctdot;</li>
+      {% elif item.number %}
+        {% if item.current %}
+          <li class="govuk-pagination__item govuk-pagination__item--current">
+            <span class="govuk-visually-hidden">Page </span>{{ item.number }}<span class="govuk-visually-hidden"> (current page)</span>
```

Comment:
I've tried applying a series of overrides for the content here, going `html > text > number`. It feels a bit complicated to me so I'm not wedded to it. It all sort of depends how attached we are to those visually hidden elements, which is the only reason to keep `number` around. I suppose we could do something clever where we default to the item's count in the array if no link text is provided in an item, but that also feels overcomplicated to me. I'm gonna defer to the team.

Reply:
I think the main reason for passing customising is for translation, where as you rightly say we can't predict where the number is going to go. Hence needing to override the entire string.

I can't really follow what you're suggesting – if you're only setting the visually hidden text then how does the visible number get defined? We don't always want to just have 1…n using the loop index as numbers as the user might be on page 140 of 300, in which case we need to be able to output `[1 … 138 139 140 141 142 … 300]`.

Having said that, your suggestion has got me wondering if we could simplify things a little by using `aria-label` – which gets rid of the user needing to pass HTML but does mean that the number is duplicated (therefore potentially allowing for a mismatch between the visible label and the visually hidden label):

```njk
<a class="govuk-link govuk-pagination__link" href="{{ item.href }}" aria-label="{{ visuallyHiddenLabel | default("Page " + item.number) }} {%- for attribute, value in item.attributes %} {{ attribute }}="{{ value }}" {%- endfor -%}>
    {{ item.number }}
</a>
```

Reply:
@lfdebrux Some general thoughts on this approach:

- I worry we're enforcing latin/romance language grammar by putting the hidden "page" before the number and not giving teams any way around this.
- We would need 2 attributes for this: `params.item.visuallyHiddenPageText` and `params.item.visuallyHiddenCurrentPageText`. The latter being for the "(current page)" after current items. We'd need to find some way to articulate when these get used which I worry is going to simply replicate the complexity of the macro in a different form.

Having a hierarchy of overrides, whilst a bit awkward as I've articulated, does solve the problem at the risk of service teams potentially doing whacky things. Something I could do to simplify this would be to remove the `text` attribute and only have `html` > `number` which defaults to the visually hidden stuff.

@36degrees Good point on the looping 🤦🏻 I do quite like the visually hidden label idea. It feels like the simplest approach in terms of the macro API whilst giving us some nice looseness. It also feels less risky than letting folks pass html which I feel is more likely to cause problems than a text only aria label. I'm gonna give that a go.

Reply:
@36degrees Do you reckon the aria label approach could apply to the prev and next buttons as well? So we could remove the need for `params.previous.html`?